crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -1,4 +1,77 @@
+//!
+//! - Check that the [`Action`] is either a [`Action::Dna`] at sequence number 0, or has a previous action with sequence number strictly greater than 0.
+//! - Checks that the author of the current action is the same as the author of the previous action.
+//! - Checks that the timestamp of the current action is greater than the timestamp of the previous action.
+//! - Checks that the sequence number of the current action is exactly 1 more than the sequence number of the previous action.
+//! - Checks that every [`Action::Create`] or [`Action::Update`] was preceeded by a [`Action::AgentValidationPkg`].
+//!
+//! ##### Store entry checks
+//!
+//! These checks are run when storing an entry that is included as part of a [`DhtOp`].
+//!
+//! - The entry type specified in the [`Action`] must match the entry type specified in the [`Entry`].
+//! - The entry hash specified in the [`Action`] must match the entry hash specified in the [`Entry`], which will be hashed as part of the check to obtain a value that is trusted.
+//! - The size of the [`Entry`] must be less than or equal to the maximum size that is accepted for this entry type. This is specified in the constant [`MAX_ENTRY_SIZE`].
+//! - If the [`Action`] is an [`Action::Update`] then the [`Update::original_action_address`] reference to the [`Action`] being updated must point to an [`Action`] that can be found locally. Once the [`Action`] address has been resolved, the [`Update::original_entry_address`] is checked against the entry address that the referenced [`Action`] specified.
+//! - If the [`Entry`] is an [`Entry::CounterSign`] then the pre-flight response signatures are checked.
+//!
+//! #### Workflow description
+//!
+//! - The workflow starts by fetching all the ops that need to be validated from the database. The ops are processed as follows:
+//!     - It then sorts the ops by [`OpOrder`], to make it more likely that incoming ops will be processed in the order they were created.
+//!     - The dependencies of these ops are then concurrently fetched from any of the local databases. Missing dependencies are handled later.
+//!     - The [validation checks](#validation-checks) are run for each op.
+//!     - For any ops that passed valdiation, they will be marked as ready for app validation in the database.
+//!     - For any ops which were rejected will be marked rejected in the database.
+//! - If any ops passed validation then app validation will be triggered.
+//! - For actions that were not found locally, the workflow will then attempt to fetch them from the network.
+//! - If any actions that were missing are found on the network, then sys validation is re-triggered to see if the newly fetched actions allow any outstanding ops to pass validation.
+//! - If fewer actions were fetched from the network than the of actions that were missing then the workflow will sleep for a short time before re-triggering itself.
```

Comment:
You could point to the sleep interval value.
```suggestion
//! - If fewer actions were fetched from the network than the actions that were missing, then the workflow will sleep for a short time before re-triggering itself.
```